File: back-end/ReportingTasksProject/01_BOL/Project.cs
using _01_BOL.Validation;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BOL
{
   public class Project
    {
        public int ProjectId { get; set; }
        [Required]
       [UniqueProjectAttribute]
        [RegularExpression("[a-zA-Z]", ErrorMessage = "only alphabet")]
        public string ProjectName { get; set; }
        [RegularExpression("[a-zA-Z]", ErrorMessage = "only alphabet")]
        [Required]
        public string ClientName { get; set; }
        [Required]
        public int TeamLeaderId { get; set; }
        [Range(0, int.MaxValue, ErrorMessage = "Please enter valid integer Number")]

        [Required]
        public int DevelopersHours { get; set; }
        [Range(0, int.MaxValue, ErrorMessage = "Please enter valid integer Number")]
        [Required]      
        public int QaHours { get; set; }
        [Required]
        [Range(0, int.MaxValue, ErrorMessage = "Please enter valid integer Number")]

        public int UiUxHours { get; set; }
        [ValidateStartDate]
        [Required]
        public DateTime StartDate { get; set; }
        [Required]
        public DateTime FinishDate { get; set; }
        public User User { get; set; }

        public bool IsActive { get; set; }


    }
}

File: back-end/ReportingTasksPhp/Models/Project.php
<?php

class Project implements JsonSerializable {

    public $ProjectId;
    public $ProjectName;
    public $ClientName;
    public $TeamLeaderId;
    public $DevelopersHours;
    public $QaHours;
    public $UiUxHours;
    public $StartDate;
    public $FinishDate;
    public $User;
    public $IsActive;

    public function __construct($sqlRaw_) {
 
        $this->ProjectId = $sqlRaw_['project_id'];
        $this->ProjectName = $sqlRaw_['project_name'];
        $this->ClientName = $sqlRaw_['client_name'];
        $this->TeamLeaderId = $sqlRaw_['team_leader_id'];
        $this->DevelopersHours = $sqlRaw_['develope_hours'];
        $this->QaHours = $sqlRaw_['qa_hours'];
        $this->UiUxHours = $sqlRaw_['ui/ux_hours'];
        $this->StartDate = $sqlRaw_['start_date'];
        $this->FinishDate = $sqlRaw_['finish_date'];
        $this->IsActive=$sqlRaw_['is_active'];
//        $this->IsActive = $sqlRaw_['is_active'];
        if(array_key_exists('user_id', $sqlRaw_))
           $this->User = new User($sqlRaw_);
//        if (array_key_exists('user_name', $sqlRaw_))


        
    }

/////היוזר מאותחל רק בid לעת עתה
    public function jsonSerialize() {
        return get_object_vars($this);
    }

}
